### PR TITLE
CLOUDP-120670 use registration config endpoint

### PR DESCRIPTION
**Pull request by tibulca**
⚠️ DO NOT MERGE UNTIL NEXT ATLAS RELEASE

## Proposed changes

_Jira ticket:_ CLOUDP-120670

Closes #120670

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by fmenezes — APPROVED** [review]
code is overall ok but I have comments

File: internal/cli/auth/register.go [hunk 1]
```diff
@@ -22,17 +22,15 @@ import (
...
 	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
 	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/oauth"
 	"github.com/spf13/cobra"
 	"go.mongodb.org/atlas/auth"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
-//go:generate mockgen -destination=../../mocks/mock_register.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/cli/auth RegisterFlow
+//go:generate mockgen -destination=../../mocks/mock_register.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/cli/auth RegisterAuthenticator,RegisterFlow
```
Comment (fmenezes) [inline internal/cli/auth/register.go:31]: the fact that we are generating mocks from this file and it does not live in the store looks suspicious, maybe a refactoring to move these to the Store would make more sense

File: internal/cli/auth/register.go [hunk 2]
```diff
@@ -71,35 +69,43 @@ func (opts *registerOpts) Run(ctx context.Context) error {
...
-	isGov         bool
-	authenticator Authenticator
+type RegisterAuthenticator interface {
+	Authenticator
+	RegistrationConfig(ctx context.Context) (*auth.RegistrationConfig, *atlas.Response, error)
 }
 
-func (ra *registerAuthenticator) RequestCode(ctx context.Context) (*auth.DeviceCode, *atlas.Response, error) {
-	// TODO:CLOUDP-121210 - Replace with new request and remove URI override.
+type registerAuthenticatorWrapper struct {
```
Comment (fmenezes) [inline internal/cli/auth/register.go:77]: [q] why do we need a Wrapper?

Comment (fmenezes) [reply internal/cli/auth/register.go:77]: answered via zoom, this is for extending the login structs

File: internal/cli/auth/register.go (hunk 1 above)
Comment (tibulca) [reply internal/cli/auth/register.go:31]: good point about having the auth calls included in the store. I will do more research here and I might include it in a list of nice to have topics for future refactoring.

**Pull request merged**